classes/cli/table.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+namespace mageekguy\atoum\cli;
+
+use mageekguy\atoum\exceptions\runtime;
+
+class table
+{
+	private $rows = array();
+	private $rowMaxLength = 0;
+	private $columnsWidth = array();
+	private $header;
+
+	public function __construct(array $header = null)
+	{
+		$this->setHeader($header);
+	}
+
+	public function __toString()
+	{
+		$columnsWidth = $this->columnsWidth;
+		$table = '';
+
+		if ($this->header !== null)
+		{
+			$header = array_pad($this->header, $this->rowMaxLength, '');
+
+			$table .= '| ' . implode(' | ',
+				array_map(
+					function($index, $column) use ($columnsWidth) {
+						return str_pad($column, $columnsWidth[$index]);
+					},
+					array_keys($header),
+					array_values($header)
+				)
+			) . ' |' . PHP_EOL;
+
+			for ($i = 0; $i < $this->rowMaxLength; $i++)
+			{
+				$table .= '|' . str_repeat('-', $this->columnsWidth[$i] + 2);
+			}
+
+			$table .= '|' . PHP_EOL;
+		}
+
+		foreach ($this->rows as $row)
+		{
+			$row = array_pad($row, $this->rowMaxLength, '');
+
+			$table .= '| ' . implode(' | ',
+				array_map(
+					function($index, $column) use ($columnsWidth) {
+						return str_pad($column, $columnsWidth[$index]);
+					},
+					array_keys($row),
+					array_values($row)
+				)
+			) . ' |' . PHP_EOL;
+		}
+
+		$this->reset();
+
+		return $table . PHP_EOL;
+	}
+
+	public function setHeader(array $header = null)
+	{
+		if ($header === null)
+		{
+			return $this;
+		}
+
+		$this->header = array_values($header);
+
+		return $this->computeRowMaxLength($this->header)->computeColumnWidths($this->header);
+	}
+
+	public function addRow(array $row)
+	{
+		$row = array_values($row);
+
+		for ($index = 1, $size = sizeof($row); $index <= $size; $index++)
```

Comment:
we should start $index at 0, and "$index < $size" should become "$index < ($size - 1)", no?